src/main/perl/Lock.pm

```diff
@@ -107,16 +107,38 @@ sub unlock
     my $self = shift;
     if ($self->{LOCK_SET}) {
         # if we forced the lock LOCK_FH can be undef
+
+        # REMOVE_LOCKFILE is set, use unlink to unlock
+        my $use_unlink = $self->{REMOVE_LOCKFILE} && -e $self->{LOCK_FILE};
+
         if ($self->{LOCK_FH}) {
-            unless (flock($self->{LOCK_FH}, LOCK_UN)) {
-                $self->error("cannot release lock: $self->{LOCK_FILE}");
-                return;
+            if (! $use_unlink) {
+                # unlock will be done by unlink. between releasing the lock here
```

Comment:
I don't follow this comment. The code below will only be run unlink if $use_unlink is true?